onnxruntime/core/providers/qnn/builder/qnn_def.h

```diff
@@ -71,6 +71,19 @@ enum class HtpPerformanceMode : uint8_t {
   kHtpExtremePowerSaver,
 };
 
+typedef struct PerThreadHtpPowerConfigs {
+  HtpPerformanceMode pre_run_perf_mode = HtpPerformanceMode::kHtpDefault;
+  HtpPerformanceMode post_run_perf_mode = HtpPerformanceMode::kHtpDefault;
+  uint32_t rpc_control_latency = 0;
+  uint32_t rpc_polling_time = 0;
+
+  bool pre_run_perf_mode_set = false;
```

Comment:
instead of an extra bool to indicate whether something is set, consider std::optional